packages/grafana-ui/src/components/Icon/iconMap.tsx

```diff
@@ -0,0 +1,259 @@
+import React from 'react';
+
+import { IconProps } from '@grafana/saga-icons';
+
+type IconImport = () => Promise<{ default: React.ComponentType<IconProps> }>;
+
+const dynamicIconImport = (iconName: string): IconImport => {
+  return () => import(/* webpackChunkName: "[request]" */ `@grafana/saga-icons/src/icons-gen/${iconName}`);
```

Comment:
I don't think this is going to work for packages that use grafana/ui outside Grafana, but those should use the new components directly (at least we can make the change in the Saga website pretty easily). Another option is to use the components from the `dist` folder, but it doesn't work in dev, probably because we set `src/index.ts` as main entry point in package.json.

Reply:
>I don't think this is going to work for packages that use grafana/ui outside Grafana, but those should use the new components directly (at least we can make the change in the Saga website pretty easily). 

There are other components that use this Icon component like Button or IconButton. Will that cause build errors too? I have a feeling this might break plugin tests and builds for any plugin that bundles grafana/ui as `src/icons-gen` won't exist.

Reply:
Good point, I'll check if it's possible to easily load them from the `dist` folder instead. An alternative is to include the `src` folder in the build, but that doesn't sound like a good idea.